usaspending_api/search/v2/views/spending_over_time.py

```diff
@@ -116,21 +119,26 @@ def apply_elasticsearch_aggregations(self, search: TransactionSearch) -> None:
         Takes in an instance of the elasticsearch-dsl.Search object and applies the necessary
         aggregations in a specific order to get expected results.
         """
-        interval = "year" if self.group == "fiscal_year" else self.group
+        interval = "year" if (self.group == "fiscal_year" or self.group == "calendar_year") else self.group
 
         # The individual aggregations that are needed; with two different sum aggregations to handle issues with
         # summing together floats.
-        group_by_time_period_agg = A(
-            "date_histogram", field="fiscal_action_date", interval=interval, format="yyyy-MM-dd"
-        )
+        group_by_time_period_agg = None
+        if self.group == "calendar_year":
+            group_by_time_period_agg = A("date_histogram", field="action_date", interval=interval, format="yyyy-MM-dd")
+        else:
+            group_by_time_period_agg = A(
+                "date_histogram", field="fiscal_action_date", interval=interval, format="yyyy-MM-dd"
+            )
         sum_as_cents_agg = A("sum", field="generated_pragmatic_obligation", script={"source": "_value * 100"})
         sum_as_dollars_agg = A(
             "bucket_script", buckets_path={"sum_as_cents": "sum_as_cents"}, script="params.sum_as_cents / 100"
         )
 
         # Putting the aggregations together; in order for the aggregations to the correct structure they
         # unfortunately need to be one after the other. This allows for nested aggregations as opposed to sibling.
-        search.aggs.bucket("group_by_time_period", group_by_time_period_agg).metric(
+        search.aggs.bucket("group_by_time_period", group_by_time_period_agg)
+        search.aggs["group_by_time_period"].bucket("group_by_category", "terms", field="award_category").metric(
```

Comment:
Not sure I follow this change. Can you explain?

Reply:
I have left a comment in the file for future devs as well but all we are doing is creating category buckets inside the group_by_time_period bucket. The `time_period` buckets we have based on the different intervals, originally we were aggregating the results of all the different award types within that bucket into one value, but now we are creating additional buckets for each of the award type. Here is an example of the buckets we are creating:

```
"group_by_time_period": {
            "buckets": [
                {
                    "key_as_string": "2007-01-01",
                    "key": 1167609600000,
                    "doc_count": 2328,
                    "group_by_category": {
                        "doc_count_error_upper_bound": 0,
                        "sum_other_doc_count": 0,
                        "buckets": [
                            {
                                "key": "contract",
                                "doc_count": 1813,
                                "sum_as_cents": {
                                    "value": 1.23815518481E11
                                },
                                "sum_as_dollars": {
                                    "value": 1.23815518481E9
                                }
                            },
                            {
                                "key": "direct payment",
                                "doc_count": 256,
                                "sum_as_cents": {
                                    "value": 3.9787618E10
                                },
                                "sum_as_dollars": {
                                    "value": 3.9787618E8
                                }
                            },
                            {
                                "key": "loans",
                                "doc_count": 124,
                                "sum_as_cents": {
                                    "value": 1.9361E7
                                },
                                "sum_as_dollars": {
                                    "value": 193610.0
                                }
                            },
                            {
                                "key": "grant",
                                "doc_count": 72,
                                "sum_as_cents": {
                                    "value": 2.6969283E9
                                },
                                "sum_as_dollars": {
                                    "value": 2.6969283E7
                                }
                            },
                            {
                                "key": "idv",
                                "doc_count": 56,
                                "sum_as_cents": {
                                    "value": 1.4634044E7
                                },
                                "sum_as_dollars": {
                                    "value": 146340.44
                                }
                            },
                            {
                                "key": "insurance",
                                "doc_count": 7,
                                "sum_as_cents": {
                                    "value": 2.086179E8
                                },
                                "sum_as_dollars": {
                                    "value": 2086179.0
                                }
                            }
                        ]
                    }
                },
```